### ISSUE TITLE
AndroidOS Investigation

### ISSUE THREAD

**Issue by veldyn** [Veldyn/2024-nas#3]
Despite narrowing down our OS to AndroidOS, there are several types of AndroidOS available. Stock android, Lineage, and other custom ROMs exist with their own benefits and drawbacks, as well as configuration options.

We decided to first look into using GrapheneOS. It's an Open Source project for Android that runs primarily on the Pixel phones, and the OS is built for security first. It makes minimal API calls, and has built-in sandboxing for all applications. This means we could still use Google services, but they would already be sandboxed. This would make meeting privacy demands extremely easy if we can get it working.

On the other hand, Graphene does not play nice with root access. Any application requesting root permissions will usually be denied, so this could be a problem down the road. However, as long as the NFS application has file permissions access, I don't see why it shouldn't work. After all, even on a traditional system there are only certain visible shared folders.

We've contacted an individual on the GrapheneOS discord to see about options for running it on a VM or on Bare Metal for testing, but they normally use a web installer so finding an ISO file could be rather tricky. We could always install the Stock GoogleOS on a Pi and then use the web installer on that, but we don't know if there are hardware restrictions / limitations that Graphene relies on that would cause issues.

If we can't get GrapheneOS to work, we'll pivot to Lineage since it provides longer-term support for functionality and security updates to android, and is functionally very similar to the stockOS. We will have to put in more work securing the system with this method however.

**Comment by veldyn** [comment]
After doing some digging we've found that GrapheneOS won't be possible. This is because Graphene relies on some existing hardware protocols in order to function securely that will be missing without the Google Pixel architecture. We could virtualize all this and try to make it work, but it's not really worth that amount of effort at the current moment.

Instead, we pivoted to LineageOS for the long term support, and stumbled across Waydroid on our way there. This seems to be an open-source Android distribution that supports a wide variety of architectures, so we'll be giving this a try as well.

We plan on spinning up two VMs, one for LineageOS and one for Waydroid and seeing how they function and testing out using NFS and SMB, and installing Samba, since these will be priority tasks for the NAS system. If we can get these running in a VM, we'll be ready to go for when we can pull the trigger on hardware purchases.

**Comment by veldyn** [comment]
It turns out that using AndroidOS may not be possible for this project. At the time of making this, there are not any supported open-source platforms for Android to host Samba on that would allow us to connect to other machines. We looked at potentially using a program like Termux to host a Linux environment within AndroidOS, but this extra layer of virtualization will decrease performance speeds, which defeats the purpose of running a lightweight OS like Android.

From here we've decided to pivot to using Ubuntu natively, and then running another OS using KVM as a bare metal VM. This will boost our security stance substantially, and allow us to use another more user-friendly operating system for the NAS. Right now we're thinking of using TrueNAS, since it is free and supports ZFS, but we are aware that there are some challenges to configuring and managing one of these systems.
